entries/workflow.qmd

```diff
@@ -1,26 +1,52 @@
 ---
 
+## GitHub and Git
+
 When adding or modifying content on either the website or the product
 itself, we follow the "branch-pull request" workflow. This is described
 in more detail in the [GitHub
 Flow](https://docs.github.com/en/get-started/quickstart/github-flow)
 page.
 
+When writing commits, follow these general guidelines:
+
+- Keep changes to files per commit as small as is reasonable and as specific as possible. For example, if making edits to the same file path across multiple files, include all those files in one commit that includes only that change to the file path.
+- Write commit messages following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) style. Use the helper VS Code extension included in our (automatically) recommended extensions to help you write these messages.
+- Write commit messages focusing on the *why* more than the *what*, though this isn't always possible.
+
+When creating branches and Pull Requests, follow these general guidelines:
+
+- Create branches following the [Conventional Branches](https://github.com/pshaddel/conventional-branch) approach by using the VSCode Extension, that is also in our (automatically) recommended extension list.
+- When creating branches and Pull Requests, keep them as focused and specific to the issue as is reasonable and limit how many changes are made in that Pull Requst. The fewer changes made in a Pull Request, the easier and faster it is for reviewers to look it over, make suggestions, and merge it in.
+- After creating a Pull Request, add it to the relevant GitHub Project.
+- As you work on your branch and it starts growing too large in scope and size, strongly consider creating "stacked Pull Requests" by making a new branch on your current working branch and submit a new Pull Request on top of the parent Pull Request branch.
+- In the Pull Request description, try to explain *why* you made the changes in the Pull Request, rather than the *what*.
+
+## Workflow helper
+
 We use some tools to help automate some of the tasks of working on
-the project repositories, such as formatting/linting Markdown and Python code or re-building a website or software. They are all found in the [`justfile`](https://just.systems/man/en/) file (see the decision post [Why Justfile](https://community.seedcase-project.org/decisions/why-justfile.html) for details on why we decided on using Justfiles).
+the project repositories, such as formatting/linting Markdown and Python code or re-building a website or software. They are all found in the [`justfile`](https://just.systems/man/en/) file.
 Using `justfile` requires opening a Terminal, and how you do that depends on the
 application you are using.
 
 > If you don't know what any of this means, the Team Lead or another
 > member will handle it.
 
-## VS Code
-
 If you use VS Code, you can run this command by using `Ctrl/Cmd-Shift-P`
 to access the Command Palette, then type out "terminal create new", and
```

Comment:
Is this just to access a terminal window, or am I missing a step?  Because in my VS Code you can use ctrl+shift+` to open and focus on a terminal window...

Reply:
@K-Beicher Do you mean the keybinding that opens an external terminal, `cmd+shift+c`? As far as I understand, there isn't a keybinding (unless you specify one yourself) for opening a new terminal within the VS Code without going through the command palette?

Reply:
@K-Beicher maybe you set that keybinding yourself? As far as I know, that isn't the default, since like @signekb I didn't have a keybinding for it either. You can do your way @K-Beicher, the reason I wrote to use the Palette is because no matter what keybinding people have, the Palette will always be the same ☺️

Reply:
@lwjohnst86 We did a bit of digging, it is an official keybinding in VS Code installed under a British Mac OS, it is referenced on VS Code's homepage about the terminal window (https://code.visualstudio.com/docs/terminal/basics). But I agree with @signekb that we should stick with things that work for all of us, so we'll go with the command pallet instructions.

Reply:
That's good to know!! So weird that it doesn't show up in the list of keybindings when you use the palette...

Reply:
Particularly as it does show in mine...  Does View: toggle terminal have a keybinding shown for you, it is ctrl+`

Reply:
@K-Beicher hmm, it does, I just didn't associate "toggle terminal" with "focus cursor at terminal". But it seems that toggling also focuses it, bit confusing!